mopidy_musicbox_webclient/static/js/functionsvars.js

```diff
@@ -176,185 +176,156 @@ function albumTracksToTable (pl, target, uri) {
     $(target).attr('data', uri)
 }
 
-function renderSongLi (song, liID, uri) {
-    var name
-    if (!song.name || song.name === '') {
-        name = uri.split('/')
-        name = decodeURI(name[name.length - 1])
+function renderSongLi (song, liID, uri, tlid, renderAlbumInfo) {
+    var name, iconClass
+    var tlidString = ''
+    var tlidParameter = ''
+    var onClick = ''
+    // Determine if the song line item will be rendered as part of an album.
+    if (!song.album || !song.album.name) {
+        iconClass = getMediaClass(song.uri)
     } else {
-        name = song.name
+        iconClass = 'trackname'
     }
-    songLi = '<li class="song albumli" id="' + liID + '">' +
-             '<a href="#" class="moreBtn" onclick="return popupTracks(event, \'' + uri + '\',\'' + song.uri + '\');">' +
+    // Play by tlid if available.
+    if (tlid) {
+        tlidString = '" tlid="' + tlid
+        tlidParameter = '\',\'' + tlid
+        onClick = 'return playTrackQueueByTlid(\'' + song.uri + '\',\'' + tlid + '\');'
+    } else {
+        onClick = 'return playTrackByUri(\'' + song.uri + '\',\'' + uri + '\');'
+    }
+    songLi = '<li class="song albumli" id="' + liID + tlidString + '">' +
+             '<a href="#" class="moreBtn" onclick="return popupTracks(event, \'' + uri + '\',\'' + song.uri + tlidParameter + '\');">' +
              '<i class="fa fa-ellipsis-v"></i></a>' +
-             '<a href="#" onclick="return playTrackByUri(\'' + song.uri + '\',\'' + uri + '\');">' +
-             '<h1 class="trackname">' + name + '</h1></a>' +
-             '</li>'
+             '<a href="#" onclick="' + onClick + '">' +
+             '<h1><i class="' + iconClass + '"></i>' + song.name + '</h1>'
+    if (renderAlbumInfo) {
+        songLi += '</p>'
+        songLi += renderSongLiTrackArtists(song)
+        if (song.album && song.album.name) {
+            songLi += ' / '
```

Comment:
Shall we switch this for a `-`? I'm not sure why it's the same separator character, it's unclear.